htsget-search/src/htsget/from_storage.rs

```diff
@@ -29,11 +30,10 @@ where
 {
   async fn search(&self, query: Query) -> Result<Response> {
     match query.format {
-      Some(Format::Bam) | None => BamSearch::new(self.storage()).search(query).await,
-      Some(Format::Cram) => CramSearch::new(self.storage()).search(query).await,
-      Some(Format::Vcf) => VcfSearch::new(self.storage()).search(query).await,
-      Some(Format::Bcf) => BcfSearch::new(self.storage()).search(query).await,
-      Some(Format::Unsupported(format)) => Err(HtsGetError::unsupported_format(format)),
```

Comment:
This change might not be as exhaustive as it was before (accounting for None/Unsupported). While I do like it to be exhaustive, the official spec only supports the formats listed ... how does this search method behave when an unsupported format is passed?

Reply:
The error gets caught here:
https://github.com/umccr/htsget-rs/blob/9eb405a5fbd993d3df87d1f78a1a4ace2baaf2d4/htsget-http-core/src/query_builder.rs#L30-L35
I think I removed it because it wasn't being used. It just moves the handling of the error outside of htsget-search, but it might make sense to keep it if htsget-search gets used with some other crate that isn't htsget-http-core?

Reply:
I think it's good practice/safe to assume that this module might/could be used by other modules in the future and therefore keep the (user) inputs tight and known (exhaustive).

Reply:
So leave it as it is right now?

Reply:
Yes, thank you!